noxfile.py

```diff
@@ -29,7 +29,15 @@ def test(session: nox.Session) -> None:
     session.run("pytest", *session.posargs)
 
 
-@nox.session()
+@nox.session(python=["3.10"])
+def type_check(session: nox.Session) -> None:
+    """Run MyPy checks."""
+    session.install("-e", ".[all]")
+    session.install("-e", ".[tests]")
+    session.run("mypy")
+
```

Comment:
Really like this this approach of reusing `nox` 👏

Reply:
If we can use the same strategy to install (test) dependencies regardless of the type of check we do, it could simplify spotting issues